Sources/MySQLKit/MySQLDataDecoder.swift

```diff
@@ -1,27 +1,22 @@
 import Foundation
+import MySQLNIO
+import NIOFoundationCompat
 
-private struct DecoderUnwrapper: Decodable {
-    let decoder: Decoder
-    init(from decoder: Decoder) {
-        self.decoder = decoder
-    }
-}
-
-private struct Wrapper<D>: Decodable where D: Decodable {
-    var value: D
-
-    init(from decoder: Decoder) throws {
-        let container = try decoder.singleValueContainer()
-        self.value = try container.decode(D.self)
-    }
-}
-
-extension MySQLData {
-    var data: Data? {
-        self.buffer.flatMap {
-            .init($0.readableBytesView)
+extension Optional: MySQLDataConvertible where Wrapped: MySQLDataConvertible {
+    public init?(mysqlData: MySQLData) {
+        if mysqlData.buffer != nil {
+            guard let value = Wrapped.init(mysqlData: mysqlData) else {
+                return nil
+            }
+            self = .some(value)
+        } else {
+            self = .none
         }
     }
+    
+    public var mysqlData: MySQLData? {
+        return self.flatMap { $0.mysqlData }
```

Comment:
Can we do `.flatMap(\.mysqlData)`? It feels a bit cleaner

Reply:
Done, nice catch! ❤️